spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/zipkin/HttpSender.java

```diff
@@ -18,37 +18,37 @@
 
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.util.Collections;
 import java.util.List;
 import java.util.zip.GZIPOutputStream;
 
-import zipkin2.Call;
-import zipkin2.CheckResult;
-import zipkin2.codec.Encoding;
 import zipkin2.reporter.BytesMessageEncoder;
 import zipkin2.reporter.ClosedSenderException;
-import zipkin2.reporter.Sender;
+import zipkin2.reporter.BytesMessageSender;
+import zipkin2.reporter.Encoding;
 
 import org.springframework.http.HttpHeaders;
 import org.springframework.util.unit.DataSize;
 
 /**
- * A Zipkin {@link Sender} that uses an HTTP client to send JSON spans. Supports automatic
+ * A Zipkin {@link BytesMessageSender} that uses an HTTP client to send JSON spans. Supports automatic
  * compression with gzip.
  *
  * @author Moritz Halbritter
  * @author Stefan Bratanov
  */
-abstract class HttpSender extends Sender {
+abstract class HttpSender extends BytesMessageSender.Base {
 
 	private static final DataSize MESSAGE_MAX_SIZE = DataSize.ofKilobytes(512);
 
+	/**
+	 * Only use gzip compression on data which is bigger than this in bytes.
+	 */
+	private static final DataSize COMPRESSION_THRESHOLD = DataSize.ofKilobytes(1);
+
 	private volatile boolean closed;
 
-	@Override
-	public Encoding encoding() {
-		return Encoding.JSON;
+	HttpSender() {
+		super(Encoding.JSON);
```

Comment:
note: the built-in senders can only work with json, as both here and also the media type header assume that. This is fine for almost everyone.